api/client.py

```diff
@@ -38,9 +38,13 @@ def listen(args):
     print("Press Ctrl-C to stop.")
 
     try:
+        # Fetch subscription ID from /subscribe endpoint response
+        subscription_id = str(res.json()['id'])
+
         while True:
-            path = '/'.join(['listen', args.channel])
-            url = urllib.parse.urljoin(args.url, path)
+            # Pass subscription ID to /listen endpoint URL
+            listen_path = '/'.join(['listen', subscription_id])
+            url = urllib.parse.urljoin(args.url, listen_path)
```

Comment:
That should work, but what I was suggesting is that the `url` could be defined outside of the `while` loop since it never changes.  It's a very minor thing though.

Reply:
Yes, that makes sense.
I will move listen_path and ```url``` outside the ```while``` loop.